### PR TITLE
Added docker helper script for publishing to container registries

### PR DESCRIPTION
**Pull request by sbayer55**
Signed-off-by: Steven Bayer <smbayer@amazon.com>

### Description
Added helper script `build-image.sh` to generate, tag, and push docker images to remote repository.
 
### Issues Resolved
closed #634
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: docker/ci/build-image.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
```
Comment (dlvenable) [inline docker/ci/build-image.sh:1]: Data Prepper already has a `release` directory. Perhaps this should go in there.

File: docker/ci/build-image.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# SPDX-License-Identifier: Apache-2.0
```
Comment (dlvenable) [inline docker/ci/build-image.sh:3]: The correct header for new code is:

```
# Copyright OpenSearch Contributors
# SPDX-License-Identifier: Apache-2.0
```

File: docker/ci/build-image.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+function usage() {
+    echo ""
+    echo "This script is used to build the Docker image with multi architecture (x64 + arm64). It prepares the files required by the Dockerfile in a temporary directory, then builds and tags the Docker image."
+    echo "--------------------------------------------------------------------------"
+    echo "Usage: $0 [args]"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-v TAG_NAME\tSpecify the image tag name such as 'centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211019'"
+    echo -e "-f DOCKERFILE\tSpecify the dockerfile full path, e.g. dockerfile/opensearch.al2.dockerfile."
```
Comment (dlvenable) [inline docker/ci/build-image.sh:19]: Is this accurate?

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: docker/ci/build-image.sh (hunk 2 above)
Comment (dlvenable) [reply docker/ci/build-image.sh:3]: Unless, this came from another project.

File: docker/ci/build-image.sh (hunk 1 above)
Comment (sbayer55) [reply docker/ci/build-image.sh:1]: I created a directory structure to match what the opensearch team is using. I can move the script to our release directory.

**Review by dlvenable — APPROVED** [review]

File: release/docker/ci/build-image.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+function usage() {
+    echo ""
+    echo "This script is used to build the Docker image. It prepares the files required by the Dockerfile in a temporary directory, then builds and tags the Docker image. Script expects to be run from the project root directory."
+    echo "--------------------------------------------------------------------------"
+    echo "Usage: $0 [args]"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-v TAG_NAME\tSpecify the image tag name such as 'centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211019'"
+    echo -e "-f DOCKERFILE\tSpecify the dockerfile full path, e.g. dockerfile/opensearch.al2.dockerfile."
```
Comment (cmanning09) [inline release/docker/ci/build-image.sh:16]: Can we have these examples relate to Data prepper?

File: release/docker/ci/build-image.sh [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    echo ""
+    echo "Optional arguments:"
+    echo -e "-h\t\tPrint this message."
+    echo "--------------------------------------------------------------------------"
+}
+
+while getopts ":hv:f:" arg; do
+    case $arg in
```
Comment (cmanning09) [inline release/docker/ci/build-image.sh:24]: The Dockerfile option (-f) will result in an invalid option with the current case statement. It looks like it is never used. Can we remove it from the usage output or is it needed?

**Review by cmanning09 — CHANGES_REQUESTED** [review]

File: release/docker/ci/build-image.sh (hunk 4 above)
Comment (sbayer55) [reply release/docker/ci/build-image.sh]: Changes made

File: release/docker/ci/build-image.sh (hunk 5 above)
Comment (sbayer55) [reply release/docker/ci/build-image.sh:23]: Changes made

**Review by dlvenable — APPROVED** [review]

**Review by cmanning09 — APPROVED** [review]

**Pull request merged**